tools/src/main/kotlin/ktx/tools/BundleLinesCreator.kt

```diff
@@ -0,0 +1,257 @@
+      .joinToString("", transform = String::capitalize)
+      .also { require(it.isNotEmpty()) { "File name $name cannot be automatically converted to a valid enum name." } }
+  }
+
+  /**
+   * Converts the name of a properties entry into a valid enum entry name.
+   * @param name The input name, retrieved from a properties file.
+   * @return A name based on [name] that is a valid enum entry name, or null if the input name cannot be used as a valid
+   * enum entry.
+   * */
+  protected open fun convertToEntryName(name: String): String? {
+    if (name.isEmpty()) {
+      println("A blank property name was encountered and will be omitted.")
+      return null
+    }
+    if ('`' in name) {
+      println("The property name \u001B[0;31m${name}\u001B[0m cannot be converted into a usable enum entry and will be omitted.")
+      return null
+    }
+    if (name[0].isDigit() || !name.all { it.isLetterOrDigit() || it == '_' })
+      return "`$name`"
+    return name
+  }
+
+  /**
+   * Generates a String representing the complete file content of a .kt file containing a `BundleLine` enum class. The
+   * default implementation also adds suppression of improper enum entry names and a companion object property that can
+   * be used to enable the `BundleLine` `nls` and `invoke` functions.
+   * @param packageName The package the enum class should be a part of.
+   * @param enumClassName The class name for the enum.
+   * @param entryNames The names of all the enum's entries.
+   * @param indent The whitespace String used to indent code.
+   * @return Source code for a complete .kt file.
+   * */
+  protected open fun generateKtFileContent(
+    packageName: String,
+    enumClassName: String,
+    entryNames: Set<String>,
+    indent: String
+  ): String {
+    return buildString {
+      append("package $packageName\n\n")
+      append("import ktx.i18n.BundleLine\n")
+      append("import com.badlogic.gdx.utils.I18NBundle\n\n")
+      append("@Suppress(\"EnumEntryName\")\n")
+      append("enum class $enumClassName : BundleLine {\n")
+      for ((index, key) in entryNames.withIndex()) {
+        val lineEnding = if (index < entryNames.size - 1) "," else ";\n"
+        append("${indent}$key$lineEnding\n")
+      }
+      append("${indent}override val bundle: I18NBundle\n")
+      append("${indent}${indent}get() = i18nBundle\n\n")
+      append("${indent}companion object {\n")
+      append("${indent}$indent/** The bundle used for [BundleLine.nls] and [BundleLine.invoke] for this enum's values. */\n")
+      append("${indent}${indent}lateinit var i18nBundle: I18NBundle\n")
+      append("$indent}\n")
+      append('}')
+    }
+  }
+}
+
+/**
+ * Parameters for running [BundleLinesCreator], intended for the `createBundleLines` Gradle task.
+ * */
+open class BundleLinesCreatorParams {
+  /**
+   * The package created enums will be placed in. Must be set, or the `createBundleLines` task cannot be used.
+   * */
+  lateinit var targetPackage: String
```

Comment:
Just to confirm, you're using an `lateinit var` because it is automatically constructed by Gradle?

Reply:
Yes. I suppose it would be less weird to make it nullable with default value of null, but call `requireNotNull` on it from the `execute` method. I think I'll do that unless you like this better.

Reply:
Either way works for me.